### PR TITLE
add olm based deploy scripts for cluster logging operator

### PR DESCRIPTION
**Pull request by bparees**
also adds e2e based on olm deployment

also leverages elasticsearch's olm deployment flow

### CODE REVIEW

**Comment by bparees** [comment]
/assign @jcantrill @ewolinetz 

note that this does not(should not) affect anything about how existing tests are run.  Once this merges i'll update the job in openshift/release to start using it (and work out any kinks via the rehearsals).

File: hack/testing-olm/assertions [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+source "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )/utils"
+assert_resources_exist(){
+	# verify deployments -- kibana, curator
+	try_until_success "oc -n $NAMESPACE get deployment kibana" ${TIMEOUT_MIN}
```
Comment (ewolinetz) [inline hack/testing-olm/assertions]: cc @igor-karpukhin this will have to change in your PR :)

File: olm_deploy/scripts/operator-uninstall.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+CLUSTER_LOGGING_OPERATOR_NAMESPACE=${CLUSTER_LOGGING_OPERATOR_NAMESPACE:-openshift-logging}
+
+
+oc delete --wait --ignore-not-found ns ${CLUSTER_LOGGING_OPERATOR_NAMESPACE}
+
+oc delete --wait --ignore-not-found crd clusterloggings.logging.openshift.io
+oc delete --wait --ignore-not-found crd collectors.logging.openshift.io
+oc delete --wait --ignore-not-found crd elasticsearches.logging.openshift.io
+oc delete --wait --ignore-not-found crd kibanas.logging.openshift.io
+oc delete --wait --ignore-not-found crd kibanas.logging.openshift.io
```
Comment (ewolinetz) [inline olm_deploy/scripts/operator-uninstall.sh]: nit, but in here twice?

**Pull request merged**